### PR TITLE
BACKLOG-11739 add SiteSwitcher component

### PR DESCRIPTION
**Pull request by Taybou**

### CODE REVIEW

File: src/javascript/DesignSystem/SiteSwitcherDisplay/SiteSwitcher.jsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
+import PropTypes from 'prop-types';
+import React, {useState} from 'react';
+import {Menu, MenuItem, withStyles} from '@material-ui/core';
+import {Button, Typography} from '@jahia/design-system-kit';
+import {ChevronDown} from 'mdi-material-ui';
+
+const styles = theme => ({
+    siteSwitcher: {
+        marginRight: `${theme.spacing.unit}!important`
```
Comment (FBerthelot) [inline src/javascript/DesignSystem/SiteSwitcherDisplay/SiteSwitcher.jsx]: why !importnant ?

File: src/javascript/DesignSystem/SiteSwitcherDisplay/SiteSwitcher.jsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+                    className={classes.siteSwitcher}
+                    onClick={handleClick}
+            >
+                <Typography noWrap variant="zeta" color="inherit">
+                    {siteNode.displayName}
+                </Typography>
+                &nbsp;
+                <ChevronDown fontSize="small" color="inherit"/>
+            </Button>
+            <Menu id="site-switcher" anchorEl={anchorEl} open={Boolean(anchorEl)} onClose={handleClose}>
```
Comment (FBerthelot) [inline src/javascript/DesignSystem/SiteSwitcherDisplay/SiteSwitcher.jsx]: Id for an element of a design system is dangerous as id must be unique for the whole page.

File: src/javascript/DesignSystem/SiteSwitcherDisplay/SiteSwitcher.jsx (hunk 1 above)
Comment (Taybou) [reply src/javascript/DesignSystem/SiteSwitcherDisplay/SiteSwitcher.jsx]: I don't know!

File: src/javascript/DesignSystem/SiteSwitcherDisplay/SiteSwitcher.jsx (hunk 2 above)
Comment (Taybou) [reply src/javascript/DesignSystem/SiteSwitcherDisplay/SiteSwitcher.jsx]: what you suggest? should I remove it?

**Pull request merged**